### PR TITLE
always run the 0-robot server on https

### LINKED ISSUE #180: Document that 0-robot should always be run on https
Since https://github.com/zero-os/0-robot/issues/171 we send user secret back to the client after each service creation, we can't send this information in plain text. So we now always must run 0-robot on top of https.

We could have the robot to auto generate signed certificate by default and add a flag to disable this feature if people deploy the 0-robot behind a proxy or something that deals with https for the robot

### PR DESCRIPTION
**Pull request by zaibon**
closes #180

### CODE REVIEW

File: utils/scripts/autobuild/node_startup.toml [hunk 1]
```diff
@@ -41,7 +41,8 @@ protected = true
 after = ["zrobot-vol"]
 
 [startup.zrobot.args]
-root = "https://hub.gig.tech/gig-autobuilder/zero-os-0-robot-autostart-0.5.1.flist"
+# root = "https://hub.gig.tech/gig-autobuilder/zero-os-0-robot-autostart-0.5.1.flist"
+root = "https://hub.gig.tech/gig-autobuilder/zero-os-0-robot-0.5.0.flist"
```
Comment (muhamadazmy) [inline utils/scripts/autobuild/node_startup.toml]: Is this intentional ? or u wanted to restore it back

Comment (zaibon) [reply utils/scripts/autobuild/node_startup.toml]: mistake indeed

**Comment by zaibon** [comment]
> Looks good, i only have one comment.
Also is it okay for the clients to accept a self-signed certificate, won't this make it vulnerable to man in the middle attacks ?

Self signed is not good I know, but I've no idea how I can do it differently to still make it secure. Any input ?

**Comment by zaibon** [comment]
won't implement ssl like this. Since for now all the robot will run only on zerotier, all communication is already p2p encrypted

**Pull request closed without merging**